### ISSUE TITLE
Wrong test grouping in Allure

### ISSUE THREAD

**Issue by rfkm** [kotest/kotest#1871]
**Which version of Kotest are you using**
4.3.1

If there are multiple tests with the same name, even if they are in different containers, they are considered to be the same test and an incorrect report will be generated. A particular problem is that even if some of them fail, the retry will show them as successful as a whole.

```kotlin
class MyTest : ShouldSpec() {
    init {
        context("context a") {
            should("work") {
                fail("boom")
            }
        }
        context("context b") {
            should("work") {
                true.shouldBeTrue()
            }
        }
    }
}
```

![image](https://user-images.githubusercontent.com/857162/100315518-1b3a4600-2ffc-11eb-8beb-82564853c537.png)
![image](https://user-images.githubusercontent.com/857162/100315531-22615400-2ffc-11eb-8518-60633be24a36.png)
![image](https://user-images.githubusercontent.com/857162/100315598-3c9b3200-2ffc-11eb-96ee-2d08a99cce8e.png)

Maybe we need to make historyId unique?
https://github.com/kotest/kotest/blob/master/kotest-extensions/kotest-extensions-allure/src/jvmMain/kotlin/io/kotest/extensions/allure/AllureWriter.kt#L87

**Comment by sksamuel** [comment]
Yep, it should be sufficient to use the fullPath from the description
rather than the test name.
Do you have time to make a PR ?

**Comment by sksamuel** [comment]
Fixed in master, will be included in 4.4.1
